File: 백준/Silver/11047. 동전 0/동전 0.py
import sys
input = sys.stdin.readline

N, K = map(int,(input().split()))
coin = []

for _ in range(N):
    coin.append(int(input()))

count = 0
for i in coin[::-1]:
    if i <= K:
        count += K // i
        K = K % i
print(count)

File: 백준/Silver/11047. 동전 0/동전 0.js
let fs = require('fs')
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n")
// let input = fs.readFileSync("./EX.txt").toString().trim().split("\n")

const info = input[0].split(" ").map(x => parseInt(x))
const N = info[0]
let K = info[1]

const A = input.slice(1, ).map(x => parseInt(x)).reverse()
cnt = 0
for (let i of A) {
    if (K >= i) {
        cnt += Math.floor(K / i) 
        K = K % i
    }
}
console.log(cnt)